atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -148,4 +149,15 @@ public long getSweepPauseMillis() {
     public int getSweepBatchSize() {
         return AtlasDbConstants.DEFAULT_SWEEP_BATCH_SIZE;
     }
+
+    @Value.Check
+    protected final void check() {
+        Preconditions.checkState(!leader().isPresent() || !lock().isPresent(),
```

Comment:
I think it's easier to parse as `!(leader().isPresent() && lock().isPresent()`.  Maybe even extract the inner part to `leaderAndLockConfigurationsAreBothPresent()`?

(Though if I remember DeMorgan's laws, this is equivalent to what you wrote)